dev-support/docker/Dockerfile

```diff
@@ -86,6 +86,15 @@ RUN apt-get -q update \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
+#######
+# Boost 1.76 (1.65 ships with Bionic)
+#######
+WORKDIR /root/boost-library
+RUN curl -L https://sourceforge.net/projects/boost/files/boost/1.72.0/boost_1_72_0.tar.bz2/download > boost_1_72_0.tar.bz2 \
+    && tar --bzip2 -xf boost_1_72_0.tar.bz2
+WORKDIR /root/boost-library/boost_1_72_0
```

Comment:
The versions are not consistent. The comment says boost 1.76 but downloads boost 1.72.

Reply:
Is there any specific reason to use the latest version of boost? If not, the version provided and maintained in Ubuntu Bionic is better.

Reply:
> The versions are not consistent. The comment says boost 1.76 but downloads boost 1.72.

Thanks for the catch @aajisaka. I've rectified it.

Reply:
> Is there any specific reason to use the latest version of boost? If not, the version provided and maintained in Ubuntu Bionic is better.

Yes @aajisaka. As part of the upgrade, I needed to use boost::system_executor - https://github.com/apache/hadoop/pull/2051/commits/4dcc89a170ec5bdb98960c20ba40d8ce05871071#diff-365634cc5fad269ad19308bd99357403R43. It's available in Boost 1.72 but not in Boost 1.65.

(Also, it saves us from having to upgrade from 1.65 to 1.7x in the future :grin:)